src/circuit-breakers.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+var CircuitBreaker = require('circuit-breaker-js');
+
+var Breaker = function() {
+	this.serviceBreakers = {};
+	this.defaultBreakerOptions = {
+		windowDuration: 10000,
+		numBuckets: 10,
+		timeoutDuration: 3000,
+		errorThreshold: 50,
+		volumeThreshold: 10
+	};
+};
+
+Breaker.prototype.instrument = function(opts) {
+	this.serviceMatchers = opts.serviceMatchers;
+	this.serviceNames = Object.keys(opts.serviceMatchers);
+
+	this.serviceNames.forEach(function(service){
+		this.serviceBreakers[service] = new CircuitBreaker(this.defaultBreakerOptions);
+
+		this.serviceBreakers[service].onCircuitOpen = function() {
+			opts.metrics.count('fetch_' + service + '_circuit_open', 1);
+		};
+		this.serviceBreakers[service].onCircuitClose = function() {
+			opts.metrics.count('fetch_' + service + '_circuit_close', 1);
+		};
+	}.bind(this));
+
+	var _fetch = GLOBAL.fetch;
+	var that = this; // not using bind as don't want to muck around with fetch's scope
+
+	GLOBAL.fetch.prototype = Object.create(_fetch.prototype);
+	GLOBAL.fetch.constructor = function(url, opts) {
```

Comment:
This bit of yucky inheritance was needed as it seems [`next-metrics`](https://github.com/Financial-Times/next-metrics/blob/master/lib/metrics/fetch.js#L73) places properties on the object which our tests depend on.